docs/basic/binary.md

```diff
@@ -48,19 +68,25 @@ The bisection method turns a problem of finding extreme values into a decision p
 
 ### Binary Search in STL
 
-One thing worth noting is that for an ordered array you can use `std::lower_bound()` to find [the first number greater than or equal to your value](http://www.cplusplus.com/reference/algorithm/lower_bound/), and `std::upper_bound()` to find the [first greater than the number of your value](https://en.cppreference.com/w/cpp/algorithm/upper_bound).
+One thing worth noting is that for an ordered array you can use `std::lower_bound()` to find [the first number greater than or equal to your value](http://www.cplusplus.com/reference/algorithm/lower_bound/), and `std::upper_bound()` to find the [first greater than the number of your value](https://en.cppreference.com/w/cpp/algorithm/upper_bound). Both of them are defined in the header `<algorithm>`.
 
-Please note that it must be an ordered array, otherwise the answer is wrong.
+Please note that it must be an ordered array, as both of them are implemented in binary search algorithm; otherwise the answer would be unexpected. <!---修正了一点点语法问题, 以及, 感觉使用Unexpected更合适-->
```

Comment:
```suggestion
Please note that it must be an ordered array, as both of them are implemented in binary search algorithm; otherwise the answer may be incorrect.
```